### PR TITLE
Updates XML documentation for the HostController

### LINKED ISSUE #2472: Docs: API XML documentation on HostController missing
## Description of problem
Many public methods on in the HostController are missing XML documentation which is very useful for developers

## Description of solution
I am doing what I can on this one now (PR submitted soon) and if there are some stuff I am not sure, I will make a list here for others to complete.

### PR DESCRIPTION
**Pull request by valadas**
Adds XML documentation for the HostController and it's interface.

Closes #2472

### CODE REVIEW

**Comment by valadas** [comment]
Most where already there on the implementation bit not on the interface and it looks like the need to be on the interface to show up in Visual Studio

File: DNN Platform/Library/Entities/Controllers/IHostController.cs [hunk 1]
```diff
@@ -26,46 +26,149 @@
...
+        /// <summary>
+        /// Takes in a <see cref="string"/> value, encrypts it with a FIPS compliant algorithm and stores
+        /// </summary>
+        /// <param name="key">host settings key</param>
+        /// <param name="value">host settings value</param>
+        /// <param name="passPhrase">pass phrase to allow encryption/decryption</param>
         void UpdateEncryptedString(string key, string value, string passPhrase);
 
+        /// <summary>
+        /// Increments the Client Resource Manager (CRM) version to bust local cache
```
Comment (ohine) [inline DNN Platform/Library/Entities/Controllers/IHostController.cs]: Bust local cache? I think we should go with `rebuild the local cache`.

Comment (valadas) [reply DNN Platform/Library/Entities/Controllers/IHostController.cs]: I don't know, isn't cache-busting the proper term here? We are not caching, we are just forcing the browser to bust the current cache, I have known this technique by that name, using a unique value in a url to bust the local cache. Google seems to agree with the term.
https://www.google.ca/search?q=cachebusting&oq=cachebusting&aqs=chrome..69i57j0l5.2238j0j7&sourceid=chrome&ie=UTF-8

File: DNN Platform/Library/Entities/Controllers/HostController.cs [hunk 2]
```diff
@@ -338,7 +342,7 @@ public void Update(string key, string value)
         }
 
         /// <summary>
-        /// takes in a text value, encrypts it with a FIPS compliant algorithm and stores
+        /// Takes in a <see cref="string"/> value, encrypts it with a FIPS compliant algorithm and stores
```
[the remark this answers is not in the document]
Comment (valadas) [reply DNN Platform/Library/Entities/Controllers/HostController.cs]: Store it I guess ;)

**Pull request merged**